src/rust/receiver-mock/src/logs.rs

```diff
@@ -115,10 +144,27 @@ fn get_timestamp_from_body(body: &str) -> Option<u64> {
     return timestamp.as_u64();
 }
 
+/// Parse the value of the X-Sumo-Fields header into a map of field name to field value
+pub fn parse_sumo_fields_header_value(header_value: &str) -> Result<LogMetadata, anyhow::Error> {
```

Comment:
This will actually be beneficial for both logs and metrics. Can you put it somewhere else so that it's not in a log specific file?

Reply:
Sure. Want me to move the header parsing out of the log-specific branch of the receiver handler as well? It'd change behaviour in that we'd respond with 400 if the header value is malformed, but that's probably ok.

Also, if I'm doing this, I should rename `LogMetadata` to just `Metadata`. Or do you prefer `Labels` like in metrics?

Reply:
> I should rename `LogMetadata` to just `Metadata`. Or do you prefer `Labels` like in metrics?

Hmm, not sure but I believe `Metadata` would fit better as a universal term for logs and metrics.

> Want me to move the header parsing out of the log-specific branch of the receiver handler as well?

I believe so and we can drop in a comment that it's unused for metrics for now.

Thanks!

Reply:
Done.